modules/internal/ChapelArray.chpl

```diff
@@ -2659,6 +2723,14 @@ module ChapelArray {
   }
 
   proc =(ref a: domain, b: domain) {
+    if a.rank != b.rank then
+      compilerError("rank mismatch in domain assignment");
+    if a.idxType != b.idxType then
+      compilerError("index type mismatch in domain assignment");
+    if isRectangularDom(a) && isRectangularDom(b) then
+      if !a.stridable && b.stridable then
+        compilerError("stridable mismatch in domain assignment");
```

Comment:
I'd like the above error message to be user-friendly. E.g. "Cannot assign from a stridable domain to an unstridable domain without an explicit cast."